### DISCUSSION TITLE
Push to repository succeeds but getting [error]Process completed with exit code 1.

### DISCUSSION THREAD

**Discussion by ercbk** [community/community#26513] [Actions]
<p>Everything in my <a href="https://github.com/ercbk/Indiana-COVID-19-Tracker/blob/master/.github/workflows/evening-update.yaml" rel="nofollow noopener">workflow</a> seems to execute correctly but I’m getting the exit code 1 error at the end which I’m guessing causes it to ostensibly fail.</p>
<p>My first attempts at the workflow did succeed, but the changes weren’t getting pushed. I was getting a message asking me who I was.</p>
<pre><code>*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'runneradmin@fv-az43.(none)')
</code></pre>
<p>I added the git config code to the workflow and now my changes get pushed to the repo, but I’m also getting failed runs. Here’s the <a href="https://pipelines.actions.githubusercontent.com/932YpfzaxGsn62tZpjWssZvRbJv0TKSQtYyqOv7xOr8b2MILvG/_apis/pipelines/1/runs/10/signedlogcontent/3?urlExpires=2020-04-15T12%3A56%3A33.6205277Z&amp;urlSigningMethod=HMACV1&amp;urlSignature=K4d0gIJz34FGedi7Lpf3xvfOv8QidxefLaKNyKEoHMM%3D" rel="nofollow noopener">log</a> to the successful run and the log to the failed <a href="https://pipelines.actions.githubusercontent.com/932YpfzaxGsn62tZpjWssZvRbJv0TKSQtYyqOv7xOr8b2MILvG/_apis/pipelines/1/runs/15/signedlogcontent/3?urlExpires=2020-04-15T12%3A55%3A18.6960468Z&amp;urlSigningMethod=HMACV1&amp;urlSignature=APVs41BkuemlyND4ijANNPMYdcOEdxkiPcKO0BnWVAg%3D" rel="nofollow noopener">one</a>. Would appreciate any assistance. I want to get that swanky passing badge. </p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/ercbk">@ercbk</a> ,</p>
<p>I can repro the same on my side. This could be due to the git command not exit cleanly on windows, you can use ‘ubunut-latest’ runner instead.</p>
<pre><code>runs-on: [ubuntu-latest]
</code></pre>
<p>Or, you can use action ‘ad-m/github-push-action@master’ instead to push your commit, code sample as below, it works on my side.</p>
<pre><code>- name: Commit files
      run: |
        git config --local user.email "action@github.com"
        git config --local user.name "GitHub Action"
        git commit -m "Add changes" -a
    - name: Push changes
      uses: ad-m/github-push-action@master
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
</code></pre>

**Comment by ercbk** [comment]
<p>It was git not exiting cleanly from Windows. Switched to Ubuntu and the run passed. Thank you very much. </p>
